components/list/src/string_matcher.rs

```diff
@@ -52,12 +52,15 @@ impl<'de: 'data, 'data> serde::Deserialize<'de> for StringMatcher<'data> {
     {
         use icu_provider::serde::borrow_de_utils::CowBytesWrap;
 
-        #[cfg(feature = "icu4x_human_readable_de")]
         if deserializer.is_human_readable() {
-            return StringMatcher::new(<&str>::deserialize(deserializer)?).map_err(|e| {
-                use serde::de::Error;
-                D::Error::custom(e.to_string())
-            });
+            use serde::de::Error;
+            #[cfg(not(feature = "serde_human"))]
+            return Err(D::Error::custom(
+                "Deserializing human-readable data formats requires the serde_human feature",
```

Comment:
nit: ".. on the icu_list crate"

(since this error may bubble out a couple crates away)

Reply:
Might be in the icu crate though...

Reply:
I moved the error up so it not only manifests for locales that have regexes (so it can be caught in testing), and I'm mentioning `ListFormatter` by name now.